shub/utils.py

```diff
@@ -183,18 +183,31 @@ def _get_egg_info(name):
     return (egg_filename, egg_path)
 
 
-def validate_jobid(jobid):
-    if not bool(re.match(r'\d+/\d+/\d+$', jobid)):
-        err = 'Job ID {} is invalid. Format should be\
-               projectid/spiderid/jobid'.format(jobid)
-        raise ClickException(err)
-
+def get_job_specs(job):
+    """
+    Parse job identifier into valid job id and corresponding API key.
 
-def get_job(jobid):
-    # XXX: Lazy import to avoid circular dependency. Needs refactoring
+    With projects default=10 and external=20 defined in config:
+    * 1/1 -> 10/1/1
+    * 2/2/2 -> 2/2/2
+    * external/2/2 -> 20/2/2
+    """
+    match = re.match(r'^((\w+)/)?(\d+/\d+)$', job)
+    if not match:
+        raise ClickException(
```

Comment:
Would it be a good idea to raise a new exception (that inherits from ClickException) here?